compiler/rustc_ast_pretty/src/pprust/state/item.rs

```diff
@@ -34,8 +34,8 @@ impl<'a> State<'a> {
         self.maybe_print_comment(span.lo());
         self.print_outer_attributes(attrs);
         match kind {
-            ast::ForeignItemKind::Fn(box ast::Fn { defaultness, sig, generics, body }) => {
-                self.print_fn_full(sig, ident, generics, vis, *defaultness, body.as_deref(), attrs);
+            ast::ForeignItemKind::Fn(func) => {
```

Comment:
Why did you change these from the exhaustive match?

Reply:
For simplicity since this is called multiple times. I can decompose the Fn item inside the `print_fn_full` so we still have an exhaustive match. What do you think?